src/package.cr

```diff
@@ -71,6 +71,28 @@ module Shards
       raise ex
     end
 
+    def install_executables
+      if installed? && spec.executables.any?
+        Dir.mkdir_p(Shards.bin_path)
+
+        spec.executables.each do |name|
+          Shards.logger.debug "Install bin/#{name}"
+          source = File.join(resolver.install_path, "bin", name)
+          destination = File.join(Shards.bin_path, name)
+
+          {% if flag?(:windows) %}
+            FileUtils.cp(source, destination)
+          {% else %}
+            if File.exists?(destination)
+              next if File.stat(destination).ino == File.stat(source).ino
+              File.delete(destination)
+            end
+            File.link(source, destination)
```

Comment:
Just because not being executed on windows doesn't mean the file system necessarily supports hard links. There should be a fallback in case `File.link` does not work.

Reply:
windows supports hard links though

Reply:
Sorry, I'm blinded. To me 1. Windows is stuck on FAT32 file systems, and 2. anything else is POSIX and supports hard links (or we're talking embedded systems and they're out-of-scope).

I'll remove the Windows check and assume POSIX until someone with better win32 knowledge ports shards.

Reply:
POSIX generally supports hard links, but not every supported filesystem does. That can be an issue for example with remote file systems.

I typically use crystal in a linux VM which mounts the windows host's file system. Creating hard links is not possible on such a file system.

There are probably other issues as well but the main point is: it doesn't need to be a hard link, a symlink or copy work as well (allthough with minor drawbacsk). Hard links are not 100% supported everywhere, so there should be an alternative as fallback.

Reply:
Sigh.